ci/common/common.sh

```diff
@@ -160,6 +174,42 @@ commit_subtree() {
   )
 }
 
+# Creates a pull request from current HEAD.
+# Current directory must be a git repo.
+#
+# ${1}: base, in github_user:branch format
+# ${2}: head, in github_user:branch format
+create_pullrequest() {
+  local base="${1:-}"
+  local head="${2:-}"
+  require_environment_variable base "${BASH_SOURCE[0]}" ${LINENO}
+  require_environment_variable head "${BASH_SOURCE[0]}" ${LINENO}
+
+  if ! check_executable hub; then
+    log_error 'create_pullrequest: "hub" not in $PATH or not executable.'
+    exit 1
+  fi
+
+  (
+    set +o xtrace
+    local rv pr_message
+
+    pr_message="$(printf '%s\n\nThis pull request is [automated](https://github.com/neovim/bot-ci).' "$(git log -1 --format='%s')")"
+
+    if ! has_gh_token ; then
+      log_info 'missing $GH_TOKEN, skipping pull-request'
+      return "$(can_fail_without_private)"
+    fi
+
+    log_info "create_pullrequest: creating pull-request ..."
+    GITHUB_TOKEN="$GH_TOKEN" hub pull-request \
```

Comment:
Unfortunately `hub` does not support `.netrc` so the token must be specified via this env var.

https://github.com/github/hub/issues/1217